src/main/java/overlap_crap_fix_locations_later/application_business/TimeTableMatchInteractor.java

```diff
@@ -0,0 +1,105 @@
+package overlap_crap_fix_locations_later.application_business;
+
+// TODO: Assuming timeTable is a list of timetableCourses. Note that the current code is kind of a standIn.
+
+import retrieve_timetable_use_case.application_business.BlockModel;
+import retrieve_timetable_use_case.application_business.CourseModel;
+import retrieve_timetable_use_case.application_business.SectionModel;
+import retrieve_timetable_use_case.application_business.TimetableModel;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class TimeTableMatchInteractor implements TimetableMatchInputBoundary {
+    /**
+     * A use case interactor responsible for finding the best overlapping TimeTable for one 'main' TimeTable, from
+     * a list of them.
+     * We seek to maximize 'overlap'. Overlap is defined in terms of:
+     * - Overlapping hours in the same section (to be with friends :3)
+     * - Possibly HARD killing all constraints OR
+     * - Possibly trying to prioritize ones where more hours maximize more constraints. LATER ON.
+     * (Weight can simply be determined by time * (proportion of satisfied constraints).
+     */
+
+    private final OverlapOutputBoundary presenter;
+    private final SectionHoursInputBoundary sectionHoursCalculator;
+
+    public TimeTableMatchInteractor(SectionHoursInputBoundary sectionHoursCalculator, OverlapOutputBoundary presenter) {
+        // Initialise what we know at compile-time. Timetables and the mainTable must be passed in later.
+        this.sectionHoursCalculator = sectionHoursCalculator;
+        this.presenter = presenter;
+    }
+
+    /**
+     * Return the overlap value of a timeTable with the main one.
+     */
+    private Double calculateTimetableOverlap(TimetableModel mainTable, TimetableModel candidate) {
+        List<CourseModel> mainCourses = mainTable.getCourses();
+        List<CourseModel> candidateCourses = candidate.getCourses();
+
+        Double totalOverlapWeightedHrs = 0.0;
+        // Although this is terribly inefficient, note that we have at most, like... 10 courses. Of 10 sections each.
+        // Section's don't store a reference to their parent course, so this is actually necessary D:
+        // Compare every section of 2 courses...
+        for (CourseModel mainCourse : mainCourses) {
```

Comment:
If we have time in the future, it would be good to extract a method or two so we don't have so much nesting going on in one function, but we'll live with this for now.

Reply:
I'll add the TODO, because you're right about this, but the presentation is in 5 hours so